sweep/sweeper.go

```diff
@@ -29,13 +64,387 @@ type UtxoSweeperConfig struct {
 	// Signer is used by the sweeper to generate valid witnesses at the
 	// time the incubated outputs need to be spent.
 	Signer lnwallet.Signer
+
+	// PublishTransaction facilitates the process of broadcasting a signed
+	// transaction to the appropriate network.
+	PublishTransaction func(*wire.MsgTx) error
+
+	// Ticker is an external dependency that drives the batch window logic.
+	Ticker ticker.Ticker
+
+	// BatchWindowSec indicates the size of the batching time window in
+	// seconds.
+	BatchWindowSec int
+
+	// SweepTxConfTarget assigns a confirmation target for sweep txes on
+	// which the fee calculation will be based.
+	SweepTxConfTarget uint32
```

Comment:
Perhaps clients should be able to express, with each sweep request, what conf bucket they'd like to be included in? Certain cases (such as CLTV sweeps) require a more timely sweep than others (CSV our force closed output).

Reply:
I don't think we need it now.

Reply:
Agreed may not be needed right now, but the current version of master has a slightly refactored form of `CreateSweepTx` that itself accept the conf target. An example of something that we may not need to confirm immediately includes a splicing transaction since we can continue to use the channel anyway. From there on if we need it to confirm ASAP, we can always try to CPFP it down via the change output.